### ISSUE TITLE
When use Bert to fine tuning coreference resolution, mention recall is high, but coreference recall is very low.

### ISSUE THREAD

**Issue by yqw-vicki** [allenai/allennlp#4852]
Please ask questions on [Stack Overflow](https://stackoverflow.com/questions/tagged/allennlp) rather than on GitHub.  We monitor and triage questions on Stack Overflow with the AllenNLP label and questions there are more easily searchable for others.
Hi, I am now using coreference resolver to retrain Bert-based coreference model, but the result is not satisfied. 
Here is my training code:
def build_trainer(
    model: Model,
    serialization_dir: str,
    train_loader: DataLoader,
    dev_loader: DataLoader
) -> Trainer:

    bert_embedder = []
    for n, p in model.named_parameters():
        if n.startswith('_text_field_embedder'):
            bert_embedder.append(n)

    parameters = [
        (n, p)
        for n, p in model.named_parameters() if p.requires_grad
    ]

    group_parameter = [
        (bert_embedder, {"lr": 2e-4})
    ]

    optimizer = AdamOptimizer(parameters, group_parameter, lr=1e-5)
    checkpoint = Checkpointer(serialization_dir=serialization_dir)
    weight_decay = MultiStepLearningRateScheduler(optimizer, list(range(0, len(train_loader.dataset.instances))))
    trainer = GradientDescentTrainer(
        model=model,
        serialization_dir=serialization_dir,
        checkpointer=checkpoint,
        data_loader=train_loader,
        validation_data_loader=dev_loader,
        patience=10,
        num_epochs=20,
        cuda_device=device,
        learning_rate_scheduler=weight_decay,
        optimizer=optimizer
    )
    return trainer

dirpath = os.path.abspath(os.path.dirname(os.getcwd()))
train_filepath = os.path.join(dirpath, "conll-formatted-ontonotes-5.0/data/train")
valid_filepath = os.path.join(dirpath, "conll-formatted-ontonotes-5.0/data/development")
transformer_model = 'bert-base-cased'
# Represents each wordpiece with an id from BERT's vocabulary.
token_indexer = PretrainedTransformerMismatchedIndexer(model_name=transformer_model, max_length=512)
reader = ConllCorefReader(30, {'bert_tokens': token_indexer}, max_sentences=10)
train_dataset = reader.read(train_filepath)
validation_dataset = reader.read(valid_filepath)
vocab = Vocabulary()
train_dataset.index_with(vocab)
validation_dataset.index_with(vocab)
# Construct a dataloader directly for a dataset which contains allennlp
# Instances which have _already_ been indexed.

train_loader = DataLoader(train_dataset, batch_size=1, collate_fn=allennlp_collate)
dev_loader = DataLoader(validation_dataset, batch_size=1, collate_fn=allennlp_collate)
embedding = PretrainedTransformerMismatchedEmbedder(transformer_model, max_length=512,
                                                    last_layer_only=True, gradient_checkpointing=True)
embedder = BasicTextFieldEmbedder(token_embedders={'bert_tokens': embedding})
encoder = LstmSeq2SeqEncoder(embedder.get_output_dim(), 200, 1, dropout=0.2, bidirectional=True)
span_dimension = 2*encoder.get_output_dim()+embedder.get_output_dim()+20
mention_feedforward = FeedForward(span_dimension, 2, [150, 150], torch.nn.ReLU())
antecedent_dimension = 3*span_dimension+20
antecedent_feedforward = FeedForward(antecedent_dimension, 2, [150, 150], torch.nn.ReLU())
corefer = CoreferenceResolver(vocab, text_field_embedder=embedder, context_layer=encoder,
                            mention_feedforward=mention_feedforward, antecedent_feedforward=antecedent_feedforward,
                            feature_size=20, max_span_width=30, spans_per_word=0.4, max_antecedents=50).to(device)

The result look like as below:
![image](https://user-images.githubusercontent.com/58431323/101536667-9a376300-399a-11eb-8f78-7fa7a3586988.png)
The coreference recall is too low even it is 15 epochs. Is there any problem in my training code?
